build: simplify handling for static variants (NFC)

compnerd opened the pull request:
There is no need to special case the products for the static build.
This handles `libdispatch_static` and `foundation_static` more
generically.

<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves SR-NNNN.

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

compnerd commented:
CC: @spevans

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please build toolchain

spevans commented on utils/build-script-impl:
```diff
@@ -1511,15 +1511,7 @@ for host in "${ALL_HOSTS[@]}"; do
-        tmp_product=${product}
-        if [[ ${tmp_product} == "libdispatch_static" ]]; then
-            tmp_product=libdispatch
-            LIBDISPATCH_STATIC_CMAKE_OPTIONS=${LIBDISPATCH_CMAKE_OPTIONS[@]}
```
Are we ok with losing this line? I not sure what it was specifically added for but if it is not an issue then this LGTM.

compnerd replied:
It seems to want to replicate the libdispatch options across to the static build.  However, that should happen implicitly further below.

compnerd committed:
build: simplify handling for static variants (NFC)

There is no need to special case the products for the static build.
This handles `libdispatch_static` and `foundation_static` more
generically.

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please smoke test
